### PR TITLE
cargo: update to serde 0.9

### PR DESCRIPTION
**Pull request by lucab**
This PR updates to latest serde version (0.9). This retains
compatibility with older rustc (<1.15) via codegen, but it can be
dropped at a later time.

### CODE REVIEW

**Comment by lucab** [comment]
Gentle ping for a review.

**Comment by arcnmx** [comment]
Sorry, been busy. FWIW, I'd lean toward just removing codegen altogether as it's no longer supported. Probably should at least change the feature names, since `unstable` and `stable` aren't really correct anymore. Maybe a `codegen` feature and have `serde_derive` included under `default`?

File: src/mod.rs [hunk 1]
```diff
@@ -1,8 +1,11 @@
-#![cfg_attr(feature = "unstable", plugin(serde_macros))]
```
Comment (arcnmx) [inline src/mod.rs:1]: The features underneath should probably be removed too?

**Comment by lucab** [comment]
I can happily switch to the new [serde_derive setup](https://serde.rs/codegen.html) if we are ok with introducing a rustc >= 1.15 contraint. Is that ok?

**Comment by arcnmx** [comment]
Yeah, considering that it's released/stable now and definitely the way forward, I'm fine with introducing that constraint.

**Comment by lucab** [comment]
Ack. I'll try to find some time to rework this, but I'm a bit constrained at the moment.

**Comment by arcnmx** [comment]
No worries, it's a minor change I'll just apply on top of this!

**Pull request merged**